framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1187,6 +1187,13 @@ private String getCharArrayStringVal(List<? extends ExpressionTree> initializers
                 } else if (e.getKind() == Tree.Kind.CHAR_LITERAL) {
                     char charVal = (((Character) ((LiteralTree) e).getValue()));
                     stringVal.append(charVal);
+                } else if (e.getKind() == Tree.Kind.IDENTIFIER) {
```

Comment:
Is there anything about this code that is specific to `IdentifierTree`? I don't think there is.

What if we made this an `else` block instead?

Further, I would also expect this code to work for `INT_LITERAL`s - so you might be able to replace the code above that handles them with this block, too.